lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -294,43 +294,50 @@ TLossFunctionUPtr createLossFunction(TLossFunctionType lossFunctionType,
 
 BOOST_AUTO_TEST_CASE(testPiecewiseConstant) {
     // Test regression quality on piecewise constant function.
+
+    auto generatePiecewiseConstant = [](TLossFunctionType lossFunctionType,
+                                        test::CRandomNumbers& rng, std::size_t cols) {
+        TDoubleVec p;
+        TDoubleVec v;
+        rng.generateUniformSamples(0.0, 10.0, 2 * cols - 2, p);
+        rng.generateUniformSamples(-10.0, 10.0, cols - 1, v);
+        for (std::size_t i = 0; i < p.size(); i += 2) {
+            std::sort(p.begin() + i, p.begin() + i + 2);
+        }
+        double offset{0.0};
+        if (lossFunctionType == TLossFunctionType::E_MsleRegression) {
+            offset = LARGE_POSITIVE_CONSTANT;
+        }
+
+        return [=](const TRowRef& row) {
+            double result{0.0};
+            for (std::size_t i = 0; i < cols - 1; ++i) {
+                if (row[i] >= p[2 * i] && row[i] < p[2 * i + 1]) {
+                    result += v[i];
+                }
+            }
+            return offset + result;
+        };
+    };
+
     test::CRandomNumbers rng;
     double noiseVariance{0.2};
     std::size_t trainRows{500};
     std::size_t testRows{200};
     std::size_t cols{3};
     std::size_t capacity{250};
+
     // TODO reactivate test for huber and MSLE
     for (auto lossFunctionType : {TLossFunctionType::E_MseRegression /*, TLossFunctionType::E_MsleRegression,
           TLossFunctionType::E_HuberRegression*/}) {
-        auto generatePiecewiseConstant = [lossFunctionType](test::CRandomNumbers& rng,
-                                                            std::size_t cols) {
```

Comment:
I was getting variable shadowing warnings here so I just moved before the setting of the variables whose names clash. And same below.